src/viam/sdk/components/movementsensor/server.cpp

```diff
@@ -0,0 +1,304 @@
+        std::dynamic_pointer_cast<MovementSensor>(rb);
+
+    AttributeMap extra;
+    if (request->has_extra()) {
+        extra = struct_to_map(request->extra());
+    }
+
+    MovementSensor::position result = movementsensor->get_position(extra);
+    *response->mutable_coordinate() = result.coordinate.to_proto();
+    response->set_altitude_m(result.altitude_m);
+
+    return ::grpc::Status();
+}
+
+::grpc::Status MovementSensorServer::GetProperties(
+    ::grpc::ServerContext* context,
+    const ::viam::component::movementsensor::v1::GetPropertiesRequest* request,
+    ::viam::component::movementsensor::v1::GetPropertiesResponse* response) {
+    if (!request) {
+        return ::grpc::Status(::grpc::StatusCode::INVALID_ARGUMENT,
+                              "Called [MovementSensor::GetProperties] without a request");
+    };
+
+    const std::shared_ptr<Resource> rb = resource_manager()->resource(request->name());
+    if (!rb) {
+        return grpc::Status(grpc::UNKNOWN, "resource not found: " + request->name());
+    }
+
+    const std::shared_ptr<MovementSensor> movementsensor =
+        std::dynamic_pointer_cast<MovementSensor>(rb);
+
+    AttributeMap extra;
+    if (request->has_extra()) {
+        extra = struct_to_map(request->extra());
+    }
+
+    MovementSensor::properties result = movementsensor->get_properties(extra);
+    response->set_linear_velocity_supported(result.linear_velocity_supported);
+    response->set_angular_velocity_supported(result.angular_velocity_supported);
+    response->set_orientation_supported(result.orientation_supported);
+    response->set_position_supported(result.position_supported);
+    response->set_compass_heading_supported(result.compass_heading_supported);
+    response->set_linear_acceleration_supported(result.linear_acceleration_supported);
+
+    return ::grpc::Status();
+}
+
+::grpc::Status MovementSensorServer::GetAccuracy(
+    ::grpc::ServerContext* context,
+    const ::viam::component::movementsensor::v1::GetAccuracyRequest* request,
+    ::viam::component::movementsensor::v1::GetAccuracyResponse* response) {
+    if (!request) {
+        return ::grpc::Status(::grpc::StatusCode::INVALID_ARGUMENT,
+                              "Called [MovementSensor::GetAccuracy] without a request");
+    };
+
+    const std::shared_ptr<Resource> rb = resource_manager()->resource(request->name());
+    if (!rb) {
+        return grpc::Status(grpc::UNKNOWN, "resource not found: " + request->name());
+    }
+
+    const std::shared_ptr<MovementSensor> movementsensor =
+        std::dynamic_pointer_cast<MovementSensor>(rb);
+
+    AttributeMap extra;
+    if (request->has_extra()) {
+        extra = struct_to_map(request->extra());
+    }
+
+    std::unordered_map<std::string, float> result = movementsensor->get_accuracy(extra);
+    response->mutable_accuracy()->empty();
+    for (const auto& i : result) {
+        response->mutable_accuracy()->emplace(i.first, i.second);
```

Comment:
Strangely this like works on my laptop but does not in CI. Different proto versions maybe?
``` 
/tmp/src/viam/sdk/components/movementsensor/server.cpp:216:39: error: no member named 'emplace' in 'google::protobuf::Map<std::basic_string<char>, float>' [clang-diagnostic-error]
        response->mutable_accuracy()->emplace(i.first, i.second);
```

Reply:
Yes, almost certainly different proto version. I remember encountering this same issue when working on the MLModel stuff and had to stop using `emplace`.